staging/src/k8s.io/code-generator/cmd/informer-gen/generators/packages.go

```diff
@@ -284,6 +284,10 @@ func factoryInterfacePackage(basePackage string, boilerplate []byte, clientSetPa
 func groupPackage(basePackage string, groupVersions clientgentypes.GroupVersions, boilerplate []byte) generator.Package {
 	packagePath := filepath.Join(basePackage, groupVersions.PackageName)
 	groupPkgName := strings.Split(string(groupVersions.Group), ".")[0]
+	// It is never valid to have a hyphen in package names, so remove all
+	// hyphens from package names to make the default path for users with
+	// group names containing hyphens better.
+	groupPkgName = strings.ReplaceAll(groupPkgName, "-", "")
```

Comment:
is this the same client-gen does?

Reply:
client-gen relies on the `--input` parameter to define the group name, which is then used as the package name. The code for this *is* reusable (see here: https://github.com/kubernetes/kubernetes/blob/7e806a2dd6efb156858706aca78f8a9850315f7b/staging/src/k8s.io/code-generator/cmd/client-gen/args/gvpackages.go)

*HOWEVER*: informer-gen doesn't accept a --input argument, so this would require quite significantly more changes to make it work/possible (incl. changing the CLI for informer-gen).

I'm not sure how we want to proceed with this 😬